database/src/models/account.rs

```diff
@@ -221,4 +223,33 @@ impl Account {
             .execute(&**db_conn)
             .and(Ok(()))
     }
+
+    pub fn display_name_or_username(&self) -> &str {
+        self.display_name.as_ref().unwrap_or(&self.username)
+    }
+}
+
+impl AsRef<Account> for Account {
+    fn as_ref(&self) -> &Account {
+        &self
+    }
+}
+
+impl<'a, 'r> FromRequest<'a, 'r> for Account {
```

Comment:
I'm somewhat uncomfortable wrt implementing this; i understand the ergonomic reasons, but there's potential in the future for wanting to be able to have one user control multiple accounts, and maybe excising this would be worse than replacing the old calls to `user.get_account(...)`? not entirely sure, though.

Reply:
I think in terms of pure code removed, whoever took on that refactor would by definition have less to remove with this route; it strikes me that the majority of work requires would be conceptual down that path, and not affected by this change.